apps/web/src/components/Spaces/Common/AvatarGrid/Avatar.tsx

```diff
@@ -0,0 +1,116 @@
+import { Image } from '@lenster/ui';
+import type { FC } from 'react';
+import React from 'react';
+import { useSpacesStore } from 'src/store/spaces';
+import { useUpdateEffect } from 'usehooks-ts';
+
+import Audio from '../Audio';
+import Dropdown from '../Dropdown';
+import CoHostData from '../Sidebar/Peers/PeerRole/CoHostData';
+import HostData from '../Sidebar/Peers/PeerRole/HostData';
+import ListenersData from '../Sidebar/Peers/PeerRole/ListenersData';
+import SpeakerData from '../Sidebar/Peers/PeerRole/SpeakerData';
+import type { IRoleEnum } from '../SpacesTypes';
+
+type Props = {
+  peerId?: string;
+  displayName: string;
+  mic?: MediaStreamTrack | null;
+  role?: IRoleEnum;
+  avatarUrl: string;
+};
+
+const Avatar: FC<Props> = ({ peerId, displayName, mic, role, avatarUrl }) => {
+  const { isMyHandRaised, setMyHandRaised, myReaction, setMyReaction } =
+    useSpacesStore();
+
+  const { me } = useHuddle01();
+
+  const RoleData = {
+    host: <HostData />,
+    coHost: peerId ? <CoHostData peerId={peerId} /> : null,
+    speaker: peerId ? <SpeakerData peerId={peerId} /> : null,
+    listener: peerId ? <ListenersData peerId={peerId} /> : null
+  } as const;
+
+  useEventListener('room:data-received', (data) => {
+    if (data.fromPeerId === peerId && data.payload['reaction']) {
+      setMyReaction(data.payload['reaction']);
+      setTimeout(() => {
+        setMyReaction('');
+      }, 5000);
+    }
+    if (
+      data.fromPeerId === peerId &&
+      (data.payload['raiseHand'] == true || data.payload['raiseHand'] == false)
+    ) {
+      setMyHandRaised(data.payload['raiseHand']);
+    }
+  });
+
+  useUpdateEffect(() => {
+    if (myReaction && peerId === me.meId) {
+      setTimeout(() => {
+        setMyReaction('');
+      }, 5000);
+    }
+  }, [myReaction]);
+
+  return (
+    <div className="relative inline-flex w-16 flex-col items-center justify-start gap-1">
+      {mic && <Audio track={mic} />}
+      <div className="group relative h-10 w-10">
+        <Image
+          src={avatarUrl}
+          className="h-10 w-10 items-center justify-center rounded-full"
+        />
+        {me.role === 'host' ||
+        (me.role === 'coHost' &&
+          role !== undefined &&
+          (me.meId === peerId || ['speaker', 'listener'].includes(role))) ||
+        ((me.role === 'speaker' || me.role === 'listener') &&
+          me.meId === peerId) ? (
+          <>
+            <div className="absolute inset-0 rounded-full group-hover:bg-black group-hover:opacity-50" />
+            <Dropdown
+              triggerChild={
+                <DotsVerticalIcon className="absolute left-1/2 top-1/2 -translate-x-1/2 -translate-y-1/2 transform text-neutral-50 opacity-0 group-hover:opacity-100" />
+              }
+            >
+              <div className="absolute -left-4 -top-5 w-40 rounded-lg border border-neutral-300 bg-white p-1 dark:border-neutral-500 dark:bg-neutral-800">
```

Comment:
why negative left?

Reply:
can you share a screenshot of this comp? cuz we are using neutral for first time here